internal/provider/remote_client.go

```diff
@@ -78,11 +109,44 @@ func (c *RemoteClient) WriteFileShell(content string, path string) error {
 		stdin.Close()
 	}()
 
-	cmd := fmt.Sprintf("cat /dev/stdin | sudo tee %s", path)
+	cmd := fmt.Sprintf("sudo touch %s", path)
+	err = run(session, cmd)
+	if err != nil {
+		return err
+	}
+
+	cmd = fmt.Sprintf("sudo chmod %s %s", permissions, path)
+	err = run(session, cmd)
+	if err != nil {
+		return err
+	}
+
+	cmd = fmt.Sprintf("cat /dev/stdin | sudo tee %s", path)
```

Comment:
Nice improvement - avoiding content within it before updating permissions!

Would be nice to group `StdinPipe` and `go func()` with the `cat /dev/stdin` that uses it. Just moving `touch` and `chmod` up before that.

Reply:
done